### DISCUSSION TITLE
Dynamic Variable from set-Cookie

### DISCUSSION THREAD

**Discussion by kdavis5** [Budibase/budibase#4719] [Help]
Hello,

I have read the documentation and believe I am following it properly but cannot seem to get this to work. Please advise if I am missing something. 

I am using a Cisco ACI APIC as a data source. It requires an authentication POST with username/Password and returns a cookie in the set-cookie field (I assume pretty standard). 

1. I send a query to authenticate. I get a 200 response and I create a dynamic variable from the returned set-cookie field. 
2. I name that "cookieACI"
3. I make a second query in the same data source with an added header, key=Cookie, and value={{cookieACI}}
4. I get an error "Error running query"

IF, I paste the cookie value from the first query directly into the second queries header value..... it works. I am testing and reviewing what Postman is doing and trying to replicate here.

IF, i add [ ] within my variable like this {{cookieACI[]}} - I do not get the error, query sends but of course the response is a 400 level. 

I am a noob with development so I hopefully am missing something small. 

Also, is there a way to force run the authentication query before other queries to ensure authentication takes place? Still reading and getting acquainted. 

Any help is greatly appreciated!

**Comment by mjashanks** [comment]
@kdavis5 - I am seeing the same issue. As far as I can see, you are doing everything correctly. Will look into getting this fixed asap

**Reply by mjashanks** [reply]
logged: https://github.com/Budibase/budibase/issues/4736

**Reply by kdavis5** [reply]
Excellent news, Thank you!
